calendar-tool/tool/src/java/org/sakaiproject/calendar/entityproviders/CalendarEventEntityProvider.java

```diff
@@ -111,23 +121,62 @@ public boolean entityExists(final String id) {
 		// user being logged in and having access to the site is handled in the
 		// API
 		Calendar cal;
-		try {
-			cal = calendarService.getCalendar(String.format(
-					"/calendar/calendar/%s/main", siteId));
+		CalendarUtil calendarUtil;
+		//check parameter for merged in request
+		if(params.get("merged") != null && (params.get("merged")).equals("true")){
+			//request is to fetch all the events from merged calendars
+			try {
+				//get the current month value for the jquery calendar in lessons
+				long timestamp = NumberUtils.toLong((String)params.get("start"), 0);
+				if(timestamp == 0){
+					//set default to current date and month
+					calendarUtil = new CalendarUtil();
+				}
+				else{
+					java.util.Calendar calendar = java.util.Calendar.getInstance();
+					calendar.setTimeInMillis(timestamp * 1000L);
+					calendarUtil = new CalendarUtil(calendar);
+				}
+				//get all events from the merged calendars
+				CalendarEventVector calendarEventVector = calendarService.getEvents(calendarService.getCalendarReferences(siteId), getMonthTimeRange(calendarUtil));
```

Comment:
This changes the default behaviour of the service again from returning all events in the calendar to returning just a months. Anyone who is using the API already and upgrades would see a change in behaviour.